invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/DeleteButton.js

```diff
@@ -8,95 +8,240 @@
  */
 
 import { i18next } from "@translations/invenio_communities/i18next";
-import React, { useEffect, useState } from "react";
-import { Button, Icon, Modal } from "semantic-ui-react";
+import React, { Component } from "react";
+import { Button, Icon, Modal, Message, Grid, Checkbox, Input } from "semantic-ui-react";
 import PropTypes from "prop-types";
+import { Trans } from "react-i18next";
+import { communityErrorSerializer } from "../../api/serializers";
+import { ErrorMessage } from "react-invenio-forms";
 
-export const DeleteButton = (props) => {
-  const [modalOpen, setModalOpen] = useState(false);
-  const [loading, setLoading] = useState(false);
-  const cancelBtnRef = React.createRef();
-  const openModalBtnRef = React.createRef();
+export class DeleteButton extends Component {
+  constructor(props) {
+    super(props);
+    this.INITIAL_STATE = {
+      modalOpen: false,
+      loading: false,
+      checkboxMembers: false,
+      checkboxRecords: false,
+      checkboxSlug: false,
+      inputSlug: "",
+      error: undefined,
+    };
+    this.checkboxRef = React.createRef();
+    this.openModalBtnRef = React.createRef();
+    this.state = this.INITIAL_STATE;
+  }
 
-  const handleOpen = () => setModalOpen(true);
+  handleInputChange = (event) => {
+    this.setState({ inputSlug: event.target.value });
+  };
+
+  handleCheckboxChange = (e, { id, checked }) => {
+    if (id === "members-confirm") {
+      this.setState({
+        checkboxMembers: checked,
+      });
+    }
+    if (id === "records-confirm") {
+      this.setState({
+        checkboxRecords: checked,
+      });
+    }
+    if (id === "slug-confirm") {
+      this.setState({
+        checkboxSlug: checked,
+      });
+    }
+  };
 
-  const handleClose = () => {
-    setModalOpen(false);
-    setLoading(false);
-    openModalBtnRef?.current?.focus();
+  handleButtonDisabled = (slug) => {
+    const { checkboxRecords, checkboxMembers, checkboxSlug, inputSlug } = this.state;
+    return !(checkboxMembers && checkboxRecords && checkboxSlug && inputSlug === slug);
   };
 
-  const handleDelete = async () => {
-    setLoading(true);
-    const { onDelete, redirectURL, onError } = props;
+  openConfirmModal = () => this.setState({ modalOpen: true });
+
+  closeConfirmModal = () => {
+    this.setState(this.INITIAL_STATE);
+    this.openModalBtnRef?.current?.focus();
+  };
+
+  handleDelete = async () => {
+    this.setState({ loading: true });
+    const { onDelete, redirectURL } = this.props;
     try {
       await onDelete();
```

Comment:
we are missing withCancel function here, which handles the case if components in being unmounted while performing the request (results with memory leaks and attempts of updating non-existing state)